tests/parser/features/test_assert_unreachable.py

```diff
@@ -0,0 +1,66 @@
+from eth_tester.exceptions import (
+    TransactionFailed,
+)
+import pytest
+
+
+def test_assure_refund(w3, get_contract):
+    code = """
+@public
+def foo():
+    assert 1 == 2, UNREACHABLE
+    """
+
+    c = get_contract(code)
+    a0 = w3.eth.accounts[0]
+    gas_sent = 10**6
+    tx_hash = c.foo(transact={'from': a0, 'gas': gas_sent, 'gasPrice': 10})
+    tx_receipt = w3.eth.getTransactionReceipt(tx_hash)
+
+    assert tx_receipt['status'] == 0
+    assert tx_receipt['gasUsed'] == gas_sent  # Drains all gains sent
+
+
+def test_basic_unreachable(w3, get_contract, assert_tx_failed):
+    code = """
+@public
+def foo(val: int128) -> bool:
+    assert val > 0, UNREACHABLE
+    assert val == 2, UNREACHABLE
+    return True
+    """
+
+    c = get_contract(code)
+
+    assert c.foo(2) is True
+
+    assert_tx_failed(lambda: c.foo(1))
+    assert_tx_failed(lambda: c.foo(-1))
+
+    with pytest.raises(TransactionFailed) as e_info:
+        c.foo(-2)
+
+    assert 'Invalid opcode 0xf' in e_info.value.args[0]
```

Comment:
```suggestion
    assert 'Invalid opcode 0xfe' in e_info.value.args[0]
```
Don't want to be generating opcodes besides 0xfe.

Reply:
How about we check the class instead:

```suggestion
    assert isinstance(e_info.__cause__, InvalidInstruction)
```

from: https://docs.python.org/3/library/exceptions.html

(I had to dig through this before and figure out that `eth-tester` is intercepting `py-evm`'s exception with the `raise new_exc from old_exc` syntax)

Reply:
That's also an OK way to do it, although do note that `InvalidInstruction` can be called with **any** invalid opcode (and could also be the result of jumping to an invalid instruction). I also think the string-based approach is, ironically, more future-proof because it would be easier to debug if upstream changes the error format :joy: .

Reply:
Why not both?